csrc/ir/utils.cpp

```diff
@@ -892,39 +892,70 @@ void validateDomainEquivalence(
       std::any_of(frontier.begin(), frontier.end(), is_symb);
   bool dom1_has_symbolic =
       std::any_of(dom1_set.begin(), dom1_set.end(), is_symb);
+
+  CompareDomainResult result;
+
+  auto check_ids = [](const auto& ids_to_check,
```

Comment:
Could you add a comment describing what this function does?

Reply:
Added